.github/workflows/test.yml

```
@@ -0,0 +1,153 @@
+
+    strategy:
+      matrix:
+        test-type: ["unit", "integration"]
+        python-version: ["3.11"]  # Use latest for faster feedback
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)